### PR TITLE
Include guards in cmake for Kokkos

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

In trying to incorporate both parthenon and another kokkos-based code into our internal codebase as submodules, I hit the snag that we don't want to build Kokkos twice. If you're working with submodules, this small change prevents that. Thanks to @AndrewGaspar for the suggestion!

Note that this is less safe than building `Kokkos` externally and then ingesting it as a library. However, if that approach is not available for whatever reason, this is a useful capability and it will help avoid cryptic linking errors.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -222,19 +222,21 @@ endif()
...
+  if (PARTHENON_IMPORT_KOKKOS)
+    find_package(Kokkos 3)
+    if (NOT Kokkos_FOUND)
+      unset(PARTHENON_IMPORT_KOKKOS CACHE)
+      message(FATAL_ERROR "Could not find external Kokkos. Consider importing a Kokkos installation into your environment or disabling external Kokkos with e.g. -DPARTHENON_IMPORT_KOKKOS=OFF")
+    endif()
   else()
-    message(FATAL_ERROR "Could not find Kokkos source. Consider running `git submodule update --init`, providing the path to a Kokkos source directory with Kokkos_ROOT, or setting PARTHENON_IMPORT_KOKKOS=ON to link to an external Kokkos installation.")
+    if (EXISTS ${Kokkos_ROOT}/CMakeLists.txt)
+      add_subdirectory(${Kokkos_ROOT} Kokkos)
```
Comment (pgrete) [inline CMakeLists.txt:234]:
```suggestion
      add_subdirectory(${Kokkos_ROOT} Kokkos)
      message(STATUS "Using Kokkos source from Kokkos_ROOT=${Kokkos_ROOT}")
```

It just occurred to me that these status messages may be helpful in cases where the enviroment/module/spack system already provides Kokkos (potentially without the user knowing).

File: CMakeLists.txt [hunk 2]
```diff
@@ -222,19 +222,21 @@ endif()
...
+    elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt)
+      add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos Kokkos)
```
Comment (pgrete) [inline CMakeLists.txt:237]:
```suggestion
      add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos Kokkos)
      message(STATUS "Using Kokkos source from Parthenon submodule at ${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos")
```

**Review by pgrete — APPROVED** [review]
Looks good :+1:

File: CMakeLists.txt (hunk 1 above)
Comment (Yurlungur) [reply CMakeLists.txt:234]: Good suggestion!

**Review by JoshuaSBrown — APPROVED** [review]
@Yurlungur Thanks for this, this is an improvement, have you tested that it does what you expect?

**Comment by Yurlungur** [comment]
> @Yurlungur Thanks for this, this is an improvement, have you tested that it does what you expect?

Thanks! I am in the process and will merge after I confirm.

**Comment by Yurlungur** [comment]
Just checked and this works as intended, so I'm merging with the two approvals. Thanks.

**Pull request merged**